Core/include/Acts/EventData/detail/GenerateParameters.hpp

```diff
@@ -106,15 +106,16 @@ inline std::pair<double, double> generateBoundDirection(
   // since we want to draw the direction uniform on the unit sphere, we must
   // draw from cos(theta) instead of theta. see e.g.
   // https://mathworld.wolfram.com/SpherePointPicking.html
-  double cosThetaMin = std::cos(options.thetaMin);
+  // Get cosThetaMin from thetaMax and vice versa
+  double cosThetaMin = std::cos(options.thetaMax);
```

Comment:
Can you elaborate?

Reply:
cos [0,pi] from should be monotonically decreasing, so the lower theta, the higher cos theta. therefore we must flip those for putting them into the distribution

Reply:
should I put this into a comment as well?

Reply:
Ah, this is only for the distribution. Yeah maybe a comment would help.